### DISCUSSION TITLE
What should the location be for compiled C-libraries under OSX?

### DISCUSSION THREAD

**Discussion by hubertus65** [modelica/ModelicaStandardLibrary#3797] [Q&A] [answered]
MLS 3.5 has the location for library binaries specified for Linux and Windows, under Resources as linux32, linux64 etc. Where should these libraries be stored for OSX, both for the INtel and the new ARM architectures? How did OpenModelica and SystemModeler do it?

**Comment by henrikt-ma** [comment, marked as answer]
This question isn't unique to MSL.  The corresponding specification issue is this one: https://github.com/modelica/ModelicaSpecification/issues/2220

**Reply by beutlich** [reply]
It also is related to FMI, where the platform identifier scheme changed in FMI v3.0.0-beta.1: https://github.com/modelica/fmi-standard/blob/master/docs/2_5_fmu_distribution.adoc#platform-tuple-examples

**Comment by henrikt-ma** [comment]
(Since you also asked, we use the name _darwin64_ for Intel binaries in SystemModeler, and as far as I know we haven't decided on anything for macOS on ARM yet – Rosetta 2 seems to do an amazingly good job.  We also use a directory called _linux-arm_.)

**Reply by sjoelund** [reply]
Why linux arm (or do you only compile for 32-bit armv7 without floating point support)? The normal name is `armhf` and FMI3 makes it somewhat clear `uname -m` names are preferred.

**Reply by henrikt-ma** [reply]
I don't think we should pay too much attention to where we currently look for things in System Modeler, but we can always hear @maltelenz has to say.

**Reply by maltelenz** [reply]
System Modeler doesn't currently support any Linux ARM architectures, so the mention in our code isn't really relevant for anything, and should definitely be ignored for the purposes of this discussion.

**Comment by sjoelund** [comment]
For amd64 Mac, OpenModelica looks in `darwin64` and `x86_64-darwin`. I guess for ARM Mac we would use whatever `uname -m` returns (with `-darwin` as suffix). I don't have access to any M1 Mac, so I wouldn't know for sure. So we look in both the locations for FMI2 (if the architecture existed in that spec) and FMI3, basically. It makes sense to use this as it is an MA standard (although, we also used it since before it was discussed in the FMI standard).

**Comment by hubertus65** [comment]
Thanks a lot to all of you, useful answers from everybody!
